jekyll/_cci2/release/configure-your-kubernetes-components.adoc

```diff
@@ -0,0 +1,225 @@
+---
+contentTags:
+  platform:
+  - Cloud
+---
+= Configure your Kubernetes components - Open preview
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Tutorial outlining how to configure your Kubernetes components for visibility and control on the releases dashboard.
+:icons: font
+:experimental:
+
+NOTE: CircleCI releases is currently in **open preview**. During open preview release features are available for free. Charges will be associated with release features once they are generally available.
+
+The steps outlined on this page guide you to configure your Kubernetes resources for integration with your release environment.
+
+[#introduction]
+== Introduction
+
+In this tutorial you will configure your Kubernetes components, adding labels and annotations to enable visibility and control over your deployments, and access to the following features:
+
+* Restore a version
+* Scale a version
+* Restart a version
+* Retry a release (when using Argo Rollouts)
+* Promote a release (when using Argo Rollouts)
+* Cancel a release (when using Argo Rollouts)
+
+[#prerequisites]
+== Prerequisites
+
+Before following the steps below, ensure the release environment has been set up successfully. Refer to the xref:set-up-a-release-environment#[Set up a release environment] page.
+
+[#add-required-labels]
+== 1. Add required labels to your Deployment/Rollout
+
+To enable your release to show up on the releases dashboard in the CircleCI web app, add the following labels to your Kubernetes Deployment or Argo Rollout:
+
+* Specify the `app` and `version` label in the Kubernetes object `Metadata.Labels`
+* Specify the `app` and `version` label in the Kubernetes object `Spec.Template.Metadata.Labels`
+
+NOTE: Do not add `version` to the selector labels. These labels can not be changed later, and `version` changes frequently.
```

Comment:
Should we be talking about the selector at all here? It just confuses the issue. We need app and version. We should specify the params of that - eg: version will change each deployment
app should remain stable
app should be under N chars
etc...

Talking about selector when we have nothing to do with selector is unnecessary, and confusing because someone who already has a deployment might think they need to have `app` in the selectors, when they dont.

Reply:
Oh interesting, thanks. Very happy to remove this section

Reply:
@ryanbarkley do you have thoughts on this? And are there any restrictions on the app label, number of characters etc. that we can list?

Reply:
We don't impose any character restrictions or string length beyond what Kubernetes itself does, so I don't think we need to get into that. @meeech you have a fair point; the reason I was suggesting this is we've had people be confused and add `version` to the selector labels when we just said "add these labels". I'm a bit torn - thoughts?

Reply:
I think it's fine to call it out. though it is not directly related to our product it might cause issues for some customers that are not very experienced with Kubernetes and that try to specify it everywhere. i don't see it as very confusing, considering we are explicitly stating that it should not be done